dags/iop/parser.py

```diff
@@ -100,3 +111,50 @@ def _get_extracted_arxiv_eprint_value(self, article):
             self.logger.error("The arXiv value is not valid.", dois=self.dois)
         except AttributeError:
             self.logger.error("No arXiv eprints found", dois=self.dois)
+
+    def _get_date_element(self, article):
+        try:
+            date = (
+                article.find("front/article-meta/history/date[@date-type='published']")
+                or article.find("front/article-meta/history/date[@date-type='epub']")
+                or article.find("front/article-meta/history/pub-date[@pub-type='ppub']")
+                or article.find("front/article-meta/pub-date")
+            )
+            return date
+        except AttributeError:
+            return
+
+    def _get_date_published(self, article):
+        date_element = self._get_date_element(article)
+        return self._get_date(date_element)
+
+    def _get_day(self, date):
+        try:
+            day = int(date.find("day").text)
+            return day
+        except (AttributeError, TypeError):
+            self.logger.error("Cannot find day of date_published in XML")
+
+    def _get_month(self, date):
+        try:
+            month = int(date.find("month").text)
+            return month
+        except (AttributeError, TypeError):
+            self.logger.error("Cannot find month of date_published in XML")
+
+    def _get_year(self, date):
+        try:
+            year = int(date.find("year").text)
+            return year
+        except (AttributeError, TypeError):
+            self.logger.error("Cannot find year of date_published in XML")
```

Comment:
maybe you can do a generic function for date extraction to which you pass an attribute name instead?

Reply:
I want to use the function **_get_year** in https://github.com/cern-sis/workflows/pull/72/files#diff-ed56f3e39a25db58f05db2a77a2915ce407ab95f5ef6635e5a4453063032e413R57 as well, that's why I separated

Reply:
I see, maybe you can derive `publication_year` from the `date_published` since it's a year part from this date? You can convert inspire https://github.com/inspirehep/inspire-utils/blob/2ae4722ce5b65c3d168b0870c2c19e03d0144954/inspire_utils/date.py#L38) and then just get the year attribute from it?

Reply:
Thanks, I can use it even for a date completion, if it's not full. On the other hand, I am not sure do I need it for year extraction. Year, month, and day are in different fields, so maybe I can just set a year as a class attrobute and easily reach it? As I did with doi:  https://github.com/cern-sis/workflows/blob/main/dags/iop/parser.py#L22